pkg/broker/util.go

```diff
@@ -43,6 +43,82 @@ func toBrokerPlans(apbPlans []apb.Plan) []Plan {
 	return brokerPlans
 }
 
+func updateMetadata(metadata map[string]interface{}, params []apb.ParameterDescriptor) map[string]interface{} {
+	if params == nil || len(params) == 0 {
+		return metadata
+	}
+
+	if metadata == nil {
+		metadata = make(map[string]interface{})
+	}
+
+	formDefinition := make([]interface{}, 0)
+
+	for paramIdx := 0; paramIdx < len(params); {
```

Comment:
I dont like the indirection of the method having the side effect of increasing the param for the loop. I also dont like that the group will increase the loop. This seems like it could be pretty tricky to debug IMO. 
Could we do this instead?
```go 
       group := map[string]interface{}{}
       groupName := ""
	for paramIDx, pd := range params {  
               	formItem, paramIdx = createUIFormItem(pd, paramIdx)
                if pd.DisplayGroup == "" && len(group) > 0 {
                     //save group to list and reset the group. 
                    // save new item to list
                } else if pd.DisplayGroup != groupName {
                    //Save group to list
                   // reset group
                   // add form item
               } else {
                  // add item to list
              }
	}
```
This just seems more clear to me, and I think (please correct me if I am wrong) will have the same behavior. @jmrodri @eriknelson @rthallisey Do you guys have strong opinions?